src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+      securityGroupIds: [securityGroup.securityGroupId],
+      storageCapacity: props.storageCapacityGiB,
+    });
+
+    return fileSystem;
+  }
+
+  /**
+   * Validates the props provided for a new FSx for Ontap file system.
+   */
+  private validateProps(props: OntapFileSystemProps) {
+    const ontapConfiguration = props.ontapConfiguration;
+    const deploymentType = ontapConfiguration.deploymentType;
+
+    this.validateHaPairs(deploymentType, ontapConfiguration.haPairs);
+    this.validateAutomaticBackupRetention(ontapConfiguration.automaticBackupRetention);
+    this.validateDailyAutomaticBackupStartTime(
+      ontapConfiguration.automaticBackupRetention,
+      ontapConfiguration.dailyAutomaticBackupStartTime,
+    );
+    this.validateDiskIops(props.storageCapacityGiB, ontapConfiguration.diskIops, ontapConfiguration.haPairs);
+    this.validateEndpointIpAddressRange(deploymentType, ontapConfiguration.endpointIpAddressRange);
+    this.validateFsxAdminPassword(ontapConfiguration.fsxAdminPassword);
+    this.validateSubnets(deploymentType, props.vpcSubnets, ontapConfiguration.preferredSubnet);
+    this.validateRouteTables(deploymentType, ontapConfiguration.routeTables);
+    this.validateThroughputCapacity(
+      deploymentType,
+      ontapConfiguration.throughputCapacity,
+      ontapConfiguration.throughputCapacityPerHaPair,
+      ontapConfiguration.haPairs,
+    );
+    this.validateStorageCapacity(ontapConfiguration.haPairs, props.storageCapacityGiB);
+  }
+
+  private validateAutomaticBackupRetention(automaticBackupRetention?: Duration): void {
+    if (
+      automaticBackupRetention == null ||
+      automaticBackupRetention.isUnresolved() ||
+      automaticBackupRetention.toMilliseconds() === 0
+    ) {
+      return;
+    }
+    if (
+      automaticBackupRetention.toMilliseconds() < Duration.days(1).toMilliseconds() ||
+      automaticBackupRetention.toDays() > 90
+    ) {
+      throw new Error('automaticBackupRetention must be between 1 and 90 days or be equal to 0');
+    }
+  }
+
+  private validateDailyAutomaticBackupStartTime(
+    automaticBackupRetention?: Duration,
+    dailyAutomaticBackupStartTime?: aws_fsx.DailyAutomaticBackupStartTime,
+  ): void {
+    if (!dailyAutomaticBackupStartTime) {
+      return;
+    }
+
+    const automaticBackupDisabled = !automaticBackupRetention || automaticBackupRetention?.toDays() === 0;
+
+    if (dailyAutomaticBackupStartTime && automaticBackupDisabled) {
+      throw new Error(
+        "'automaticBackupRetention' period must be set a non-zero day when 'dailyAutomaticBackupStartTime' is set",
+      );
+    }
+  }
+
+  private validateDiskIops(storageCapacityGiB: number, diskIops?: number, haPairs: number = 1): void {
+    if (
+      diskIops == null ||
+      Token.isUnresolved(diskIops) ||
+      Token.isUnresolved(storageCapacityGiB) ||
+      Token.isUnresolved(haPairs)
+    ) {
+      return;
+    }
+
+    const minDiskIops = storageCapacityGiB * 3 * haPairs;
```

Comment:
This validation is based on these description.

> The minimum and maximum values for this property depend on the value of HAPairs and StorageCapacity. The minimum value is calculated as StorageCapacity * 3 * HAPairs (3 IOPS per GB of StorageCapacity). The maximum value is calculated as 200,000 * HAPairs.


https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-diskiopsconfiguration.html#cfn-fsx-filesystem-diskiopsconfiguration-iops